modules/_labs/pagination/react/lib/Pagination/GoTo/useGoToForm.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import {useUniqueId} from '@workday/canvas-kit-react-common';
+
+import {PaginationModel} from '../types';
+
+export interface UseGoToFormConfig extends PaginationModel {
+  inputId?: string;
+  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
+}
+
+export const useGoToForm = (
+  {inputId, onSubmit, state, events}: UseGoToFormConfig = {} as UseGoToFormConfig
+) => {
+  const [value, setValue] = React.useState<number>();
+
+  React.useEffect(() => {
+    if (value !== undefined && state.currentPage !== value) {
+      setValue(state.currentPage);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [state.currentPage]);
```

Comment:
The Compound Component Model API would make `onSetCurrentPage` be fired whenever the page changes. This model would then handle state updates instead of using `useEffect` which prevents disabling of `exhaustive-deps`. You could also achieve this by changing `inputProps`:

```tsx
const inputProps = {
  id: uniqueId,
  value: (value !== undefined && state.currentPage !== value) ? state.currentPage : value,
  onChange: handleChange,
}
```

Either will prevent double rendering from the `setValue` call inside a `useEffect`

Reply:
I'm not sure that provided example will work. You'd need to do some additional state tracking to know when `state.currentPage` was updated and `value` was not.

Reply:
> You'd need to do some additional state tracking to know when `state.currentPage` was updated and value was not.

Yeah, that's why the `useEffect` makes sense here, imo. But I understand the double-rendering concern. I'll mess with it.

Reply:
I think the model solution is best. I'm not seeing a clear way to track the `state.currentPage` changes.